### DISCUSSION TITLE
Learning rewards using graph neural networks

### DISCUSSION THREAD

**Discussion by akpas001** [pyg-team/pytorch_geometric#3455] [Q&A]
I need to train a Gnn based on the data of pre existing next_states and rewards. next states are one hot encoded and rewards are random numbers(can be negative as well as float). I have constructed a gnn using GINConv.

According to Markov's decision process in Reinforcement learning, a state_action pair as an input to the network gives the next_state and reward.  (state, action, next_states are one-hot encoded).

So, i am forming a state action pair inside the network and trying to calculate next state and reward loss. the loss graph for next states is good but the graph for rewards is very bad(rewards are not learning). I donot understand what mistake i am doing here. Can somebody help me?

Here I am attaching the snippets of the code, and graphs of next_state loss and reward loss respectively. Can somebody help me with that? 

![1](https://user-images.githubusercontent.com/56883247/140769475-d950a306-9e82-4486-a886-d1611d147c80.PNG)
![2](https://user-images.githubusercontent.com/56883247/140769479-79a5fb26-b531-470c-ad9d-275b716722a1.PNG)
![3](https://user-images.githubusercontent.com/56883247/140769481-e98755f5-81ba-4dec-aa06-bb2b11e0e657.PNG)
![4](https://user-images.githubusercontent.com/56883247/140769483-47bcec0e-ec11-4818-9f9e-ba1c5ea3d963.PNG)
